File: typescript/packages/bench/src/rules/partCostMarginRule.ts
/** Cardinality fixture: nominal (class, `implements`) implementor of RuleContract. */

import type { RuleContract } from "../contracts/ruleContract";
import type { Dataset } from "../dataset";

export class PartCostMarginRule implements RuleContract {
    readonly key = "part-cost-margin";

    evaluate(data: Dataset): boolean {
        return data.partsCostCents() > 0;
    }
}

File: rust/crates/atelier-bench/src/rules/part_cost_margin_rule.rs
//! Rule 19/24 (textual half).

use crate::contracts::rule_contract::RuleContract;
use crate::dataset::Dataset;

pub struct PartCostMarginRule;

impl RuleContract for PartCostMarginRule {
    fn evaluate(&self, data: &Dataset) -> bool {
        data.parts_cost_cents() >= 0
    }
}
